src/ivc.py

```diff
@@ -0,0 +1,656 @@
+    In-vessel component outer surface Cartesian point-loci data is contained
+    within the Data.radial_build dictionary.
+
+    Arguments:
+        vmec (object): plasma equilibrium VMEC object from PyStell-UW.
+        build (dict): dictionary defining stellarator build. See Stellarator
+            class docstring for more detail.
+        repeat (int): number of times to repeat build segment.
+        num_phi (int): number of phi geometric cross-sections to make for each
+            build segment (defaults to 61).
+        num_theta (int): number of points defining the geometric cross-section
+            (defaults to 61).
+        scale (double): a scaling factor between the units of VMEC and [cm].
+        plasma_h5m_tag (str): optional alternate material tag to use for
+            plasma. If none is supplied and the plasma is not excluded,
+            'plasma' will be used.
+        sol_h5m_tag (str): optional alternate material tag to use for
+            scrape-off layer. If none is supplied and the scrape-off layer is
+            not excluded, 'sol' will be used.
+        logger (object): logger object.
+    '''
+
+    def __init__(
+            self,
+            vmec,
+            build,
+            repeat,
+            num_phi,
+            num_theta,
+            scale,
+            plasma_h5m_tag,
+            sol_h5m_tag,
+            logger
+    ):
+        self.vmec = vmec
+        self.build = build
+        self.repeat = repeat
+        self.num_phi = num_phi
+        self.num_theta = num_theta
+        self.scale = scale
+        self.plasma_h5m_tag = plasma_h5m_tag
+        self.sol_h5m_tag = sol_h5m_tag
+        self.logger = logger
+
+        self.phi_list = np.deg2rad(self.build['phi_list'])
+        try:
+            assert self.phi_list[0] == 0.0, \
+                'Initial toroidal angle not equal to 0. Please redefine ' \
+                'phi_list, beginning at 0.'
+        except AssertionError as e:
+            self.logger.error(e.args[0])
+            raise e
+
+        self.theta_list = np.deg2rad(self.build['theta_list'])
+        try:
+            assert self.theta_list[-1] - self.theta_list[0] == 2*np.pi, \
+                'Poloidal extent is not 360 degrees. Please ensure poloidal ' \
+                'angles are specified for one full revolution.'
+        except AssertionError as e:
+            self.logger.error(e.args[0])
+            raise e
+
+        self.wall_s = self.build['wall_s']
+        self.radial_build = self.build['radial_build']
+
+        self.generate_data()
+
+    def generate_data(self):
+        '''Generates data defining stellarator in-vessel component geometry.
+        '''
+        n_phi = len(self.phi_list)
+        n_theta = len(self.theta_list)
+
+        # Extract toroidal extent of build
+        self.seg_tor_ext = self.phi_list[-1]
+        self.tot_tor_ext = (self.repeat + 1)*self.seg_tor_ext
```

Comment:
The toroidal extent is also defined independently for the source mesh. I found that keeping the geometry and source toroidal extent definitions independent was useful since one may want to repeat the segment geometry in OpenMC (or other MC code) rather than ParaStell for improved performance. In this case, the source still needs to span the full geometry used in MC, which would be different from the geometry modeled in ParaStell